src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/PluginSourceRepository.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using Microsoft.Performance.Toolkit.Plugins.Core.Discovery;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Common;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime
+{
+    /// <summary>
+    ///    Represents a repository of <see cref="PluginSource"/>s.
+    /// </summary>
+    public sealed class PluginSourceRepository
```

Comment:
If we change the interface to observable collection, we could ditch this class and just use `ObservableCollection<PluginSource>`